graph_snyk_cve_sync.py

```diff
@@ -312,14 +282,128 @@ def _extract_data_from_feed(self):
             elif eco == "python" and self._parse_data_for_eco(eco):
                 logger.info("Parsing feed for Pypi.")
                 self._parse_data(self.snyk_data[eco], "pypi")
+            elif eco == "golang" and self._parse_data_for_eco(eco):
+                logger.info("Parsing feed for Golang.")
+                self._parse_golang_data(self.snyk_data[eco], "golang")
             else:
                 logger.info("Ignoring the ecosystem {} from the feed".format(eco))
 
     def _is_date_in_range(self, date):
         """Check to see if the date of vuln falls in the range."""
         date_obj = datetime.strptime(date.split('T')[0], '%Y-%m-%d')
+        """When running under delta feed mode, we need to consider only those vulns which were
+        updated between the given offset date and today's date."""
         return self.today > date_obj >= self.start_day
 
+    def _generate_default_cve_obj(self, eco, pkg, versions, latest, gh=None, lic=None):
+        """Generate the default cve object."""
+        # Generate the default snyk cve object.
+        obj = {
+            "affected": [],
+            "vulnerabilities": [],
+            "all_ver": versions,
+            "latest_version": latest,
+            "ecosystem": eco,
+            "package": pkg
+        }
+        # 2 extra fields are needed in case of golang for github url and license details
+        if eco == "golang":
+            obj['gh_link'] = gh
+            obj['license'] = lic
+        return obj
+
+    def _parse_golang_data(self, vuln_data, eco):
```

Comment:
I see a lot of common code, can this be refactored into common methods. Also this method is very long and should be split.

Reply:
@prashbnair yes there are few repeated lines of code. Only problem is that the entire flow is not the same as other ecosystem. embedding it all in one function was causing a lot of confusion and if else sort of logic.
Hence i removed it to a separate function.

Reply:
Problem with splitting is, our maintainability index, which increases as soon as i add a new function. That will cause build failure..i know it sounds strange, but thats how it behaves

Reply:
My opinion is this should be refactored. I understand the need. for a separate function for goLang but remove all the common code into separate methods, probably a Parser class also makes sense.  I do not understand why the maintainability index is complaining.

Reply:
Have removed all the possible common code. Cannot reduce it further due to different logic between golang and other eco systems.